hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -264,6 +268,31 @@ public AbfsConfiguration getAbfsConfiguration() {
     return this.abfsConfiguration;
   }
 
+  public void access(Path path, FsAction mode) throws IOException {
+    LOG.debug("access for filesystem: {}, path: {}, mode: {}",
+        this.client.getFileSystem(), path, mode);
+    if (!this.abfsConfiguration.isCheckAccessEnabled()) {
+      return;
+    }
+    String relativePath = path.isRoot() ? "" : ("/" + getRelativePath(path));
+    throwFileNotFoundIfNotExists(relativePath);
```

Comment:
We should not do GetFileStatus call. Response for checkaccess call should be the source of response to client.